qutebrowser/components/braveadblock.py

```diff
@@ -0,0 +1,321 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2020 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Functions related to ad blocking."""
+
+import io
+import os.path
+import logging
+import typing
+import pathlib
+
+from PyQt5.QtCore import QUrl
+
+from qutebrowser.api import (
+    hook,
+    config,
+    message,
+    downloads,
+    interceptor,
+    apitypes,
+    qtutils,
+)
+from qutebrowser.api.interceptor import ResourceType
+from qutebrowser.components.utils import blockutils
+
+try:
+    import adblock
+except ImportError:
+    adblock = None  # type: ignore[assignment]
+
+
+logger = logging.getLogger("network")
+ad_blocker = typing.cast(typing.Optional["BraveAdBlocker"], None)
```

Comment:
I tried this and I get
```
qutebrowser/components/braveadblock.py:281: error: Statement is unreachable  [unreachable]
            ad_blocker.update_files()
            ^
qutebrowser/components/braveadblock.py:289: error: Statement is unreachable  [unreachable]
            ad_blocker.enabled = _should_be_used()
```

Reply:
Works fine for me. Are you sure you had the `type:` at the start of the comment so that mypy picks it up? Those messages sound like mypy thinks it will always be `None`

Reply:
I think you were right that I forgot the `type:` prefix. Thanks.